pkg/clustermesh/common/remote_cluster.go

```diff
@@ -267,7 +267,7 @@ func (rc *remoteCluster) getClusterConfig(ctx context.Context, backend kvstore.B
 	rc.config = &models.RemoteClusterConfig{Required: true}
 	rc.mutex.Unlock()
 
-	cfgch := make(chan types.CiliumClusterConfig)
+	cfgch := make(chan types.CiliumClusterConfig, 1)
```

Comment:
> Let's fix this by making the channel buffered, to ensure that the controller never blocks while attempting to send to it.

Generally speaking this is an incorrect assumption. Buffered channels still block once the channel is full.

I haven't looked into how the channel is used, but I want to raise this for discussion to make sure we're really addressing the underlying weakness and not just making it into an even more weird rare deadlock case.

Reply:
> Generally speaking this is an incorrect assumption. Buffered channels still block once the channel is full.

Sure thing. However, that channel is guaranteed to be sent to at most once, as used to communicate that the configuration has been correctly retrieved by the controller. Upon success, the controller body is never executed again, as `RunInterval` is set to 0.

https://github.com/cilium/cilium/blob/1c4d0fdf3f6da9d98da0d7259392483f31a3d963/pkg/clustermesh/common/remote_cluster.go#L279-L296

An alternative possibility could be refactoring this logic to not use the controller. That would simplify a bit the flow, at the cost of losing the the observability capabilities automatically provided by the controllers infrastructure. 

Differently, using an unbuffered channel in combination with `select` would not be correct, and expose a different race condition if sending to the channel before starting to read from it.